Document interaction of asynchronous operations with thread scopes (#143)
NVIDIA/libcudacxx: docs/extended_api/asynchronous_operations.md

## Asynchronous Operations

> Asynchronous operations are performed _as-if_ in other threads. This other threads are related to the thread requesting the asynchronous operations via all scope relationships [^thread_scope].

Author: ```suggestion
Asynchronous operations are potentially performed in a new thread. This new threads are related to the thread requesting the asynchronous operations via all scope relationships [^thread_scope].
```
